### PR TITLE
Fix/add cucumber tests

### PR DESCRIPTION
**Pull request by Rhian**
cucumber

This PR contains all the changes from the `feature/improve-dashboard` branch, rebased against the `fix/new-sections` branch, and with Cucumber tests for the admin dashboard added.

### CODE REVIEW

File: features/step_definitions/dashboard_steps.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  click_button("Mark as 'on Delta'")
+end
+
+Then('the {string} column contains {string}') do |_string, _string2|
+  within(".delta") do
+    expect(page).to have_content("Yes")
+  end
+end
+
+Given('I try to sign in with bad credentials') do
```
Comment (freesteph) [inline features/step_definitions/dashboard_steps.rb:50]: This feels like a single-use step, could we maybe use the other ones in web_steps? https://github.com/unboxed/southwark-bsp/blob/fix/new-sections/features/step_definitions/web_steps.rb

File: features/step_definitions/dashboard_steps.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  end
+end
+
+When('I mark building as {string}') do |_string|
+  page.find("#building_building_id_").check
+  click_button("Mark as 'on Delta'")
+end
+
+Then('the {string} column contains {string}') do |_string, _string2|
+  within(".delta") do
```
Comment (freesteph) [inline features/step_definitions/dashboard_steps.rb:45]: same as above

File: app/javascript/packs/check_all.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+function checkAll() {
```
[the remark this answers is not in the document]
Comment (Rhian) [reply app/javascript/packs/check_all.js:1]: Now it waits for the doc to load (Turbolinks removed to sync with your branch), then looks for the element and attaches the listener there 👍

File: app/views/admin/buildings/_flash_content.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      </div>
+    </div>
+  <% else %>
+    <div class="govuk-!-margin-top-6 govuk-!-margin-bottom-8">
+      <div class="govuk-error-summary__body govuk-!-padding-2">
+        <svg class="moj-banner__icon" color="green" role="presentation" focusable="false" xmlns="http://www.w3.org/2000/svg"
+             viewBox="0 0 25 25" height="25" width="25">
+          <path d="M25,6.2L8.7,23.2L0,14.1l4-4.2l4.7,4.9L21,2L25,6.2z"></path>
+        </svg>
+        <%= msg %>
```
Comment (freesteph) [inline app/views/admin/buildings/_flash_content.html.erb:23]: What's special about the "not updated" error? I can't see the difference besides a bullet point.

This is the flash on the side of the form:

```erb
<% if @survey.errors.any? %>
  <div class="govuk-error-summary">
    <h2 class="govuk-error-summary__title">
      There is a problem
    </h2>
    <div class="govuk-error-summary__body">
      <ul class="govuk-list">
        <% @survey.errors.full_messages.each do |message| %>
          <li>
            <span class="govuk-error-message"><%= message %></span>
          </li>
        <% end %>
      </ul>
    </div>
  </div>
<% end %>
```

File: features/step_definitions/dashboard_steps.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+Given('I try to sign in with bad credentials') do
+  visit "/admin"
+
+  fill_in 'Email address', with: "fake@example.com"
+  fill_in 'Password', with: "unknown"
+  click_button 'Sign in'
+end
+
+Then('I should not see the dashboard content') do
+  expect(page).not_to have_content("Dashboard")
```
[the remark this answers is not in the document]
Comment (Rhian) [reply features/step_definitions/dashboard_steps.rb:49]: This was actually really useful as it meant I spotted a spelling error in the notice. But I decided to also leave in the step about not seeing "Dashboard" as we need to ensure the user cannot view content, as well as seeing the warning

File: features/admin_interacts_with_dashboard.feature [hunk 6]
```diff
@@ -0,0 +1,18 @@
+Feature: Admin views and edits buildings on dashboard
+  Scenario: Admin can see relevant content
+    Given I am on the dashboard
+    Then the dashboard contains all expected columns
+    And the dashboard contains the expected building information
```
Comment (freesteph) [inline features/admin_interacts_with_dashboard.feature:5]: I find this line quite vague so I'd actually remove the step "`And(the dashboard contains[...]`) and put the expectations directly in here.

File: features/step_definitions/dashboard_steps.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    And there is a building in the system
+    And I log in
+  )
+end
+
+Given('I am on the sign in page') do
+  visit "/admin"
+end
+
+And('there is a building in the system') do
```
Comment (freesteph) [inline features/step_definitions/dashboard_steps.rb:13]: We have a very similar step that could probably be extracted and reused: https://github.com/unboxed/southwark-bsp/blob/fix/new-sections/features/step_definitions/survey_steps.rb#L11

**Review by freesteph — APPROVED** [review]
Happy with it :palm_tree: 

I've left some minor comments but you can address them at your discretion.

File: features/step_definitions/dashboard_steps.rb (hunk 7 above)
Comment (Rhian) [reply features/step_definitions/dashboard_steps.rb]: I didn't spot this! Good suggestion -> less code :)

File: features/admin_interacts_with_dashboard.feature (hunk 6 above)
Comment (Rhian) [reply features/admin_interacts_with_dashboard.feature]: Done!

File: app/views/admin/buildings/_flash_content.html.erb (hunk 4 above)
Comment (Rhian) [reply app/views/admin/buildings/_flash_content.html.erb:23]: I'll defer this one until after the other bulk actions are implemented as we might need something else entirely.

**Pull request merged**